🐛 CI failure: 'functools.partial' has no attribute 'submodules' in test_paged_stashing

**Describe the bug**

CI tests in `tests/unit_tests/transformer/moe/test_paged_stashing.py` are failing with an `AttributeError` because the test treats a `functools.partial` as if it were a `ModuleSpec`:

```
tests/unit_tests/transformer/moe/test_paged_stashing.py:139: AttributeError
E   AttributeError: 'functools.partial' object has no attribute 'submodules'
```

Failing nodes:

- `TestPagedStashing::test_forward_backward_4_layers`
- `TestPagedStashingOverBudget::test_overload_factor_and_over_budget`

Tag @NVIDIA/mcore-oncall to get oncall's attention to this issue.

**Root cause (likely)**

At `tests/unit_tests/transformer/moe/test_paged_stashing.py:139` the test does:

```python
transformer_layer_spec = get_gpt_layer_with_transformer_engine_spec(
    num_experts=self.config.num_moe_experts, moe_grouped_gemm=True
)
...
MoELayer(self.config, transformer_layer_spec.submodules.mlp.submodules)
```

In the MoE branch, `transformer_layer_spec.submodules.mlp` is set to `partial(MoELayer, submodules=MoESubmodules(...))` (see `megatron/core/models/gpt/moe_module_specs.py:66`, introduced by [#3435](https://github.com/NVIDIA/Megatron-LM/pull/3435) on 2026-05-10). A `functools.partial` does not expose `.submodules`, so the attribute access at line 139 crashes.

Commit history:

- `5e31514165` ([#3435](https://github.com/NVIDIA/Megatron-LM/pull/3435), 2026-05-10, @nick-schank) switched MoE spec to return a `partial`.
- `f007db77b9` ([#4247](https://github.com/NVIDIA/Megatron-LM/pull/4247), 2026-05-22, @nanz-nv) added the test that assumes the old `ModuleSpec` shape.

**Failing run**

| Field | Value |
|-------|-------|
| PR    | [#4931: test: enable NVTE_CUTEDSL_FUSED_GROUPED_MLP via pytest fixture](https://github.com/NVIDIA/Megatron-LM/pull/4931) (surfaced here; the PR itself does not touch paged stashing) |
| Run   | [26290097610](https://github.com/NVIDIA/Megatron-LM/actions/runs/26290097610) |
| Job   | [`tests/unit_tests/transformer/moe/**/*.py - latest`](https://github.com/NVIDIA/Megatron-LM/actions/runs/26290097610/job/77389398728) |

**Error (verbatim, abridged)**

```
tests/unit_tests/transformer/moe/test_paged_stashing.py:139: AttributeError
E   AttributeError: 'functools.partial' object has no attribute 'submodules'

FAILED tests/unit_tests/transformer/moe/test_paged_stashing.py::TestPagedStashing::test_forward_backward_4_layers
FAILED tests/unit_tests/transformer/moe/test_paged_stashing.py::TestPagedStashingOverBudget::test_overload_factor_and_over_budget
```

**Steps/Code to reproduce bug**

Re-run the failing CI job linked above, or locally inside the dev container:

```bash
pytest tests/unit_tests/transformer/moe/test_paged_stashing.py
```

**Additional context**

Triaged automatically via `/create-issue`. Fix is to update `_create_moe_layer` to construct `MoESubmodules` directly (or to read it via `.keywords["submodules"]` on the partial) rather than chaining `.submodules.mlp.submodules`.

--- a/tests/unit_tests/transformer/moe/test_paged_stashing.py
+++ b/tests/unit_tests/transformer/moe/test_paged_stashing.py
@@ -208,7 +208,6 @@ class TestPagedStashing:
     def teardown_method(self, method):
         Utils.destroy_model_parallel()
 
-    @pytest.mark.flaky_in_dev
     @pytest.mark.skipif(not torch.cuda.is_available(), reason="CUDA not available")
     @pytest.mark.internal
     def test_forward_backward_4_layers(self):
@@ -298,7 +297,6 @@ class TestPagedStashingOverBudget:
     def teardown_method(self, method):
         Utils.destroy_model_parallel()
 
-    @pytest.mark.flaky_in_dev
     @pytest.mark.skipif(not torch.cuda.is_available(), reason="CUDA not available")
     @pytest.mark.internal
     def test_overload_factor_and_over_budget(self):

--- a/tests/unit_tests/transformer/moe/test_paged_stashing.py
+++ b/tests/unit_tests/transformer/moe/test_paged_stashing.py
@@ -196,6 +196,19 @@ _TE_GROUPED_MLP_OP_FUSER_SKIP_REASON = (
 )
 
 
+def _is_mxfp8_supported() -> bool:
+    """MXFP8 quantization in TE requires compute capability >= 10.0 (Blackwell)."""
+    if not torch.cuda.is_available():
+        return False
+    return torch.cuda.get_device_capability()[0] >= 10
+
+
+_MXFP8_SKIP_REASON = (
+    "MXFP8 (tests configure fp8_recipe='mxfp8') requires compute capability >= 10.0 (Blackwell)"
+)
+
+
+@pytest.mark.skipif(not _is_mxfp8_supported(), reason=_MXFP8_SKIP_REASON)
 @pytest.mark.skipif(
     not _te_grouped_mlp_op_fuser_environment_supported(),
     reason=_TE_GROUPED_MLP_OP_FUSER_SKIP_REASON,
@@ -285,6 +298,7 @@ class TestPagedStashing:
             ), f"tokens_per_expert != ref: max diff = {(tpe_f - ref_f).abs().max().item()}"
 
 
+@pytest.mark.skipif(not _is_mxfp8_supported(), reason=_MXFP8_SKIP_REASON)
 @pytest.mark.skipif(
     not _te_grouped_mlp_op_fuser_environment_supported(),
     reason=_TE_GROUPED_MLP_OP_FUSER_SKIP_REASON,

